LobbyManager.h

```diff
@@ -1,44 +1,45 @@
 #ifndef TICTACTOE_SERVER_LOBBYMANAGER_H
 #define TICTACTOE_SERVER_LOBBYMANAGER_H
 
-#include "SearchManager.h"
+#include "PlayerInfo.h"
 #include <unordered_map>
 #include <random>
+#include <stdexcept>
 
 class LobbyManager {
 public:
 	struct Lobby {
-		SearchManager::PlayerInfo player1;
-		SearchManager::PlayerInfo player2;
+		PlayerInfo player1;
+		PlayerInfo player2;
 		bool player1_active;
 		bool player2_active;
 		unsigned search_id;
 
-		Lobby(SearchManager::PlayerInfo player1, SearchManager::PlayerInfo player2, unsigned id) : player1(player1), player2(player2), search_id(id){}
+		Lobby(PlayerInfo player1, PlayerInfo player2, unsigned lobby_id) : player1(player1), player2(player2), search_id(lobby_id){}
```

Comment:
This is a naming mismatch. Why are you passing `lobby_id` to a field named `search_id`?